.github/workflows/dluhc-build-and-publish.yml

```diff
@@ -0,0 +1,213 @@
+      #     path: ${{matrix.app}}/test-results
+      #     retention-days: 14
+
+      # - name: Upload test coverage artifacts
+      #   uses: actions/upload-artifact@v2
+      #   if: ${{ success() || failure() }}
+      #   with:
+      #     name: test-coverage-${{matrix.app}}
+      #     path: ${{matrix.app}}/test-coverage
+      #     retention-days: 14
+
+  setup-env-vars:
+    needs: lint-and-test
+    runs-on: ubuntu-latest
+    outputs:
+      BRANCH_NAME: ${{steps.set-vars.outputs.LOWER_CASE_BRANCH}}
+    steps:
+      - id: set-vars
+        run: |
+          echo "::set-output name=LOWER_CASE_BRANCH::$(
+            echo ${GITHUB_REF#refs/heads/} |
+            tr "[:upper:]" "[:lower:]"
+          )"
+      - name: echo-vars
+        id: echo-vars
+        run: |
+          echo ${{steps.set-vars.outputs.LOWER_CASE_BRANCH}}
+          echo $VERSION
+
+  build-and-push-images:
+    # needs: [lint-and-test, setup-env-vars]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        app: [designer, runner]
+    name: build-and-push-${{matrix.app}}-image
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: "12.x"
+      - name: Docker compose pull
+        run: docker-compose pull
+      - name: Create .env for ${{ matrix.app }} workspace
+        run: |
+          touch ./${{ matrix.app }}/.env
+          echo LAST_TAG_GH=$VERSION >> ./${{ matrix.app }}/.env
+          echo LAST_COMMIT_GH=${{ github.sha }} >> ./${{ matrix.app }}/.env
+          cat ./${{ matrix.app }}/.env
+      # - name: Docker compose build ${{ matrix.app }}
+      #   run: |
+      #     LAST_TAG='$VERSION'
+      #     LAST_COMMIT='${{ github.sha }}'
+      #     docker-compose build ${{ matrix.app }}
+
+      - name: Docker metadata
+        id: metadata
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{env.IMAGE_REPO_PATH}}/${{env.IMAGE_NAME_STUB}}${{ matrix.app }}
+          tags: |
+            type=sha,format=long
+            type=raw,value=latest,enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+            type=raw,value=${{env.VERSION}},enable=${{ github.ref == format('refs/heads/{0}', 'main') }}
+            type=ref,event=branch
+
+      # - name: Log in to registry
+      #   run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u $ --password-stdin
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v1
+        with:
+          registry: ${{ env.DOCKER_REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push docker image
+        uses: docker/build-push-action@v2
+        with:
+          context: ${{matrix.app}}
+          tags: ${{ steps.metadata.outputs.tags}}
+          labels: ${{ steps.metadata.outputs.labels }}
+          push: true
+          file: Dockerfile
```

Comment:
Looks like it is failing because this path doesn't exist. Did we mean to use the runner Dockerfile here or something else?

Reply:
Looks like previous this was based on the app matrix? e.g. https://github.com/communitiesuk/digital-form-builder/blob/97be2fe917d7c08856c1c37c62509e5260829b80/.github/workflows/dluhc-build-and-publish.yml#L114

Reply:
Thanks - fixed the path but that version of the build-push-action didn't like subfolders so I've increased it to v3 and it works.